http/go/nethttp/redirect_handler.go

```diff
@@ -0,0 +1,140 @@
+package nethttplibrary
+
+import (
+	"errors"
+	nethttp "net/http"
+	"net/url"
+	"strings"
+
+	abs "github.com/microsoft/kiota/abstractions/go"
+)
+
+type RedirectHandler struct {
+	options RedirectHandlerOptions
+}
+
+func NewRedirectHandler() *RedirectHandler {
+	return NewRedirectHandlerWithOptions(RedirectHandlerOptions{
+		MaxRedirects: DEFAULT_MAX_REDIRECTS,
+		ShouldRedirect: func(req *nethttp.Request, res *nethttp.Response) bool {
+			return true
+		},
+	})
+}
+func NewRedirectHandlerWithOptions(options RedirectHandlerOptions) *RedirectHandler {
+	return &RedirectHandler{options: options}
+}
+
+type RedirectHandlerOptions struct {
+	ShouldRedirect func(req *nethttp.Request, res *nethttp.Response) bool
+	MaxRedirects   int
+}
+
+var redirectKeyValue = abs.RequestOptionKey{
+	Key: "RedirectHandler",
+}
+
+type redirectHandlerOptionsInt interface {
+	abs.RequestOption
+	GetShouldRedirect() func(req *nethttp.Request, res *nethttp.Response) bool
+	GetMaxRedirect() int
+}
+
+func (o *RedirectHandlerOptions) GetKey() abs.RequestOptionKey {
+	return redirectKeyValue
+}
+func (o *RedirectHandlerOptions) GetShouldRedirect() func(req *nethttp.Request, res *nethttp.Response) bool {
```

Comment:
Use more descriptive name like **handler**

Reply:
it's a getter and name is following conventions, unless Go has different conventions I'm not aware of?

Reply:
I am not familiar with the convention you're talking about but it's general wisdom to use readable variables.

Reply:
I am referring to the **o** variable name

Reply:
aaah the conversation makes much more sense to me now 😂
thanks, fixed.